src/novika/packages/impl/colors.cr

```diff
@@ -16,14 +16,14 @@ module Novika::Packages::Impl
       {ri.to_u8, gi.to_u8, bi.to_u8}
     end
 
-    def with_color_echo(engine, fg : Color?, bg : Color?, form : Form)
+    def with_color_echo(engine, sys : ISystem, fg : Color?, bg : Color?, form : Form)
```

Comment:
I do think it's bad to pass it as an arg here because the use of system is an implementation detail and must be explicitly such.